### PR TITLE
requester: Provide a button to "auto-lowercase" HTTP/2 header names

### PR DESCRIPTION
**Pull request by amitpanwar789**
## Overview
I have added getButtonLowerCaseHeaderName JButton in requester requestPanel.
which simple lowerCase all the headers name for HTTP/2 protocol

## Related Issues
Fix zaproxy/zaproxy#8176
@thc202 please review this

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [ ] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 1]
```diff
@@ -255,6 +265,27 @@ private JToggleButton getButtonHostHeader() {
...
+        if (lowerCaseHeaderName == null) {
+            lowerCaseHeaderName =
+                    new JButton(ExtensionRequester.createIcon("lowercase-header-button.png"));
+            lowerCaseHeaderName.setToolTipText("Lowercase Headers");
+            lowerCaseHeaderName.addActionListener(e -> lowerCaseHeaderNameButtonTriggered());
+        }
+        return lowerCaseHeaderName;
+    }
+
+    private void lowerCaseHeaderNameButtonTriggered() {
```
Comment (kingthorin) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:278]:
```suggestion
    private void lowerCaseHeaderNamed() {
```

**Review by kingthorin — COMMENTED** [review]
The help and changelog should be updated as well.

**Comment by kingthorin** [comment]
You should squash or fixup the commits and start using amending commits instead of adding more. In the end we should be merging a single commit.

You'd do this via git's interactive rebase and forced pushing, if you need instructions let us know.

**Comment by amitpanwar789** [comment]
can you tell me how to squash the commits, i  tried like git rebase HEAD~5 but, i think something wrong here check last commit

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 2]
```diff
@@ -262,20 +263,23 @@ private JToggleButton getButtonHostHeader() {
...
+                    Constant.messages.getString(
+                            "requester.httpsender.checkbox.lowerCaseHeadersName"));
+            lowerCaseHeaderNames.addActionListener(e -> lowerCaseHeaderNamed());
         }
-        return lowerCaseHeaderName;
+        return lowerCaseHeaderNames;
     }
 
-    private void lowerCaseHeaderNameButtonTriggered() {
+    private void lowerCaseHeaderNamed() {
```
Comment (kingthorin) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:282]: Oops did I  typo that? Should have been "s" not "d".

File: addOns/requester/src/main/resources/org/zaproxy/addon/requester/images/lowercase-header-button.png
[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/resources/org/zaproxy/addon/requester/images/lowercase-header-button.png:1]: it is fine to me , i can see the icon without this error

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/resources/org/zaproxy/addon/requester/images/lowercase-header-button.png:1]: Okay, I’ll try on another device

**Comment by psiinon** [comment]
We try to keep to the Diagona and Fugue Icon sets.
Fugue doesnt have an uppercase to lower icon but it has the reverse, so I hacked it 😉 

edit-lowercase

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 3]
```diff
@@ -255,6 +265,29 @@ private JToggleButton getButtonHostHeader() {
...
+    }
+
+    private void lowerCaseHeaderNames() {
+        customHttpPanelRequest.saveData();
+        HttpRequestHeader httpRequestHeader =
+                ((HttpMessage) customHttpPanelRequest.getMessage()).getRequestHeader();
+        for (HttpHeaderField field : httpRequestHeader.getHeaders()) {
+            httpRequestHeader.setHeader(field.getName(), null);
+            httpRequestHeader.addHeader(field.getName().toLowerCase(Locale.ROOT), field.getValue());
+        }
```
[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:287]: I gave it a try and it appears to be working as expected. Would you mind sharing some additional information?

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:287]: Try with two fields with the same name.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:287]: The least amount of change/deviation from original. The user/code may have specifically wanted/needed two. So to answer your question the message should still have two lowercase headers with the same name.

[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: ok done

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Unit test should still be added though.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: I had a quick look at that and couldn't sort out the mocking. In particular `CustomHttpPanelRequest` seems to be a weird cases that mockito claims it or it's supers aren't reachable.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Something like this?

```diff
$ git diff HEAD
diff --git a/addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java b/addOns/requester/src/main/java/o
rg/zaproxy/addon/requester/internal/HttpPanelSender.java
index 77e343974..97e993119 100644
--- a/addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java
+++ b/addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java
@@ -292,6 +292,17 @@ public class HttpPanelSender {
         customHttpPanelRequest.updateContent();
     }

+    static void testLowerCaseHeaderNames(HttpMessage msg) {
+        HttpRequestHeader httpRequestHeader = msg.getRequestHeader();
+        List<HttpHeaderField> fields = httpRequestHeader.getHeaders();
+        for (HttpHeaderField field : fields) {
+            httpRequestHeader.setHeader(field.getName(), null);
+        }
+        for (HttpHeaderField field : fields) {
+            httpRequestHeader.addHeader(field.getName().toLowerCase(Locale.ROOT), field.getValue());
+        }
+    }
+
     /**
      * A {@link HttpRedirectionValidator} that enforces the {@link
      * org.parosproxy.paros.control.Control.Mode Mode} when validating the {@code URI} of
diff --git a/addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java b/addOns/requester/src/tes
t/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java
new file mode 100644
index 000000000..0c6d5618a
--- /dev/null
+++ b/addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java
@@ -0,0 +1,44 @@
+package org.zaproxy.addon.requester.internal;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+
+import org.apache.commons.httpclient.URI;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.network.HttpHeaderField;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+class HttpPanelSenderUnitTest {
+
+    private static final String URI = "http://example.com";
+    private HttpMessage msg;
+
+    @BeforeEach
+    void init() throws IOException {
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI(URI, false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+    }
+
+    @Test
+    void shouldLowercaseheaderNames() {
+        // Given
+        msg.getRequestHeader().addHeader("X-Foo", "bar");
+        msg.getRequestHeader().addHeader("X-Client", "Foo-bar 1.1.0");
+        // When
+        HttpPanelSender.testLowerCaseHeaderNames(msg);
+        // Then
+        assertThat(msg.getRequestHeader().getHeaders().size(), is(equalTo(2)));
+        assertThat(msg.getRequestHeader().getHeaders(),
+                containsInAnyOrder(new HttpHeaderField("x-foo", "bar"),
+                        new HttpHeaderField("x-client", "Foo-bar 1.1.0")));
+    }
+}
```

[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: I'm learning about unit testing but don't have much experience. Could you recommend some resources or sample tests for reference? Thanks!

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Assuming @thc202 or someone else agrees you can apply the above diff/patch then simply add a test that checks the duplicate header situation.

You can also definitely read through tests in any of the add-ons to get ideas/guideance.

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: The new method would be called by the existing though, but yeah, that looks good.

[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: I am wondering if I need to create a new test method like testLowerCaseHeaderNames in the HttpPanelSender.java file to test the lowerCaseHeaderNames function. Additionally, I noticed that the Requester class appears to have very few unit tests, with only RequesterParamUnitTest.java being present. why is it so ?

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Yes as outlined/agreed above.

It’s short on tests because it’s old code.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: > The new method would be called by the existing though, but yeah, that looks good.

@amitpanwar789 this still needs to be done, maybe instead of test* call it set* and use it from the button method (then the code is t duplicated). The method probably also doesn't need "public" visibility, try just "static void".

[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: but we need to make it public otherwise i am getting error in HttpPanelSenderUnitTest file "'setLowerCaseHeaderNames(org.parosproxy.paros.network.HttpMessage)' is not public in 'org.zaproxy.addon.requester.internal.HttpPanelSender'. Cannot be accessed from outside package"

```
    public void lowerCaseHeaderNames() {
        customHttpPanelRequest.saveData();
        HttpMessage msg = ((HttpMessage) customHttpPanelRequest.getMessage());
        setLowerCaseHeaderNames(msg);
        customHttpPanelRequest.updateContent();
    }

    static void setLowerCaseHeaderNames(HttpMessage msg) {
        HttpRequestHeader httpRequestHeader = msg.getRequestHeader();
        List<HttpHeaderField> fields = httpRequestHeader.getHeaders();
        for (HttpHeaderField field : fields) {
            httpRequestHeader.setHeader(field.getName(), null);
        }
        for (HttpHeaderField field : fields) {
            httpRequestHeader.addHeader(field.getName().toLowerCase(Locale.ROOT), field.getValue());
        }
    }

```

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Okay that's fine, I wasn't totally sure it would work that's why I said "try" 😁

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: It does not need to be public, the problem is that the test class is not in the same package as the class being tested.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: Good catch. @amitpanwar789 if you add the unit test class to `intetnal` (a new folder) then it should work without being public.

[the remark this answers is not in the document]
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java]: ok

File: addOns/requester/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 
 
+## Added 
+- Added Button to LowerCase Headers Name
```
Comment (kingthorin) [inline addOns/requester/CHANGELOG.md:11]: The blanks at lines 8 and 9 can be removed and just one after the note. The note should be a sentence (add a period, fix the caps, make name plural) and you should add an issue reference.

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 5]
```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
         return hostHeader;
     }
 
+    private JButton getButtonLowerCaseHeaderName() {
```
Comment (kingthorin) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:269]: Name should be plural on this method too

**Review by kingthorin — COMMENTED** [review]
Looking good

File: addOns/requester/CHANGELOG.md [hunk 6]
```diff
@@ -5,7 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
-
+## Added 
+- Added Button to lowercase Header Names
```
Comment (kingthorin) [inline addOns/requester/CHANGELOG.md:9]: > Button to lowercase request header names (Issue 8176).

Comment (amitpanwar789) [reply addOns/requester/CHANGELOG.md]: Completed, and I truly appreciate all your assistance. Thank you!

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 7]
```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
         return hostHeader;
     }
 
+    private JButton getButtonLowerCaseHeaderNames() {
+        if (lowerCaseHeaderNames == null) {
+            lowerCaseHeaderNames =
+                    new JButton(ExtensionRequester.createIcon("lowercase-header-button.png"));
+            lowerCaseHeaderNames.setToolTipText(
+                    Constant.messages.getString(
+                            "requester.httpsender.button.lowerCaseHeadersName"));
```
Comment (thc202) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:275]: lowerCaseHeaderNames

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 8]
```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
...
+            lowerCaseHeaderNames.addActionListener(e -> lowerCaseHeaderNames());
+        }
+        return lowerCaseHeaderNames;
+    }
+
+    private void lowerCaseHeaderNames() {
+        customHttpPanelRequest.saveData();
```
Comment (thc202) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:282]: Should catch exceptions and warn/inform the user (see how it behaves when changing the method with a malformed request header).

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 9]
```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
...
+        HttpMessage msg = ((HttpMessage) customHttpPanelRequest.getMessage());
```
Comment (thc202) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:283]: Remove extra parentheses.

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java [hunk 10]
```diff
@@ -255,6 +266,36 @@ private JToggleButton getButtonHostHeader() {
...
+        setLowerCaseHeaderNames(msg);
+        customHttpPanelRequest.updateContent();
```
Comment (thc202) [inline addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:285]: IMO this could be done in the lambda and have the following method called`lowerCaseHeaderNames`.

File: addOns/requester/src/main/resources/org/zaproxy/addon/requester/Messages.properties [hunk 11]
```diff
@@ -22,6 +22,7 @@ requester.httppanel.methodchange = Method
 requester.httppanel.methodchange.warn = Unable to change the method, failed to set the data to the message.
 requester.httppanel.tab.request = Request
 requester.httppanel.tab.response = Response
+requester.httpsender.button.lowerCaseHeadersName = LowerCase Header Names
```
Comment (thc202) [inline addOns/requester/src/main/resources/org/zaproxy/addon/requester/Messages.properties:25]: Lowercase

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    void init() throws IOException {
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI(URI, false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+    }
+
+    @Test
+    void shouldLowercaseheaderNames() {
```
Comment (thc202) [inline addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java:50]: `…CaseHeader…`

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        // Given
+        msg.getRequestHeader().addHeader("X-Foo", "bar");
+        msg.getRequestHeader().addHeader("X-Client", "Foo-bar 1.1.0");
+        msg.getRequestHeader().addHeader("X-Client", "Foo-not-bar 2.0");
+        // When
+        HttpPanelSender.setLowerCaseHeaderNames(msg);
+        // Then
+        assertThat(msg.getRequestHeader().getHeaders().size(), is(equalTo(3)));
```
Comment (thc202) [inline addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java:58]: Unnecessary when using `contains*` (contains also gives a better error message indicating which ones are missing or are unexpected).

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+        assertThat(
+                msg.getRequestHeader().getHeaders(),
+                containsInAnyOrder(
```
Comment (thc202) [inline addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java:61]: `contains`, the order matters.

**Comment by thc202** [comment]
The help needs to be updated.

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java (hunk 10 above)
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:285]: are you telling me to name the method to lowercase the header name as lowerCaseHeaderNames, but we have JButton name's same as lowerCaseHeaderNames , is this not cause any problem ?

Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:285]: A method with that name already exists, I'm suggesting using that name for the other method and have this logic in the lambda.

File: addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java (hunk 8 above)
Comment (amitpanwar789) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:282]: does it look good now ? 

I have also added this in Messages.properties "requester.httppanel.lowercaseHeaderNames.warn = Unable to lowercase the header names, failed to set the data to the message."
```

    private JButton getButtonLowerCaseHeaderNames() {
        if (lowerCaseHeaderNames == null) {
            lowerCaseHeaderNames =
                    new JButton(ExtensionRequester.createIcon("lowercase-header-button.png"));
            lowerCaseHeaderNames.setToolTipText(
                    Constant.messages.getString(
                            "requester.httpsender.button.lowerCaseHeaderNames"));
            lowerCaseHeaderNames.addActionListener(
                    e -> {
                        try {
                            customHttpPanelRequest.saveData();
                        } catch (InvalidMessageDataException er) {
                            StringBuilder warnMessage = new StringBuilder(150);
                            warnMessage.append(
                                    Constant.messages.getString(
                                            "requester.httppanel.lowercaseHeaderNames.warn"));

                            String exceptionMessage = er.getLocalizedMessage();
                            if (exceptionMessage != null && !exceptionMessage.isEmpty()) {
                                warnMessage.append('\n').append(exceptionMessage);
                            }
                            View.getSingleton().showWarningDialog(warnMessage.toString());
                            return;
                        }
                        HttpMessage msg = (HttpMessage) customHttpPanelRequest.getMessage();
                        lowerCaseHeaderNames(msg);
                        customHttpPanelRequest.updateContent();
                    });
        }
        return lowerCaseHeaderNames;
    }

    static void lowerCaseHeaderNames(HttpMessage msg) {
        HttpRequestHeader httpRequestHeader = msg.getRequestHeader();
        List<HttpHeaderField> fields = httpRequestHeader.getHeaders();
        for (HttpHeaderField field : fields) {
            httpRequestHeader.setHeader(field.getName(), null);
        }
        for (HttpHeaderField field : fields) {
            httpRequestHeader.addHeader(field.getName().toLowerCase(Locale.ROOT), field.getValue());
        }
    }
```

Comment (thc202) [reply addOns/requester/src/main/java/org/zaproxy/addon/requester/internal/HttpPanelSender.java:282]: I think it does.

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java [hunk 15]
```diff
@@ -52,10 +50,7 @@ void shouldLowercaseheaderNames() {
         msg.getRequestHeader().addHeader("X-Foo", "bar");
         msg.getRequestHeader().addHeader("X-Client", "Foo-bar 1.1.0");
         msg.getRequestHeader().addHeader("X-Client", "Foo-not-bar 2.0");
-        // When
-        HttpPanelSender.setLowerCaseHeaderNames(msg);
-        // Then
-        assertThat(msg.getRequestHeader().getHeaders().size(), is(equalTo(3)));
+        HttpPanelSender.lowerCaseHeaderNames(msg);
```
Comment (thc202) [inline addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java:53]: The When/Then should be kept though.

**Comment by amitpanwar789** [comment]
> The help needs to be updated.

can you please, tell me more about help you are talking, which i need to update!

**Comment by kingthorin** [comment]
addOns/requester/src/main/javahelp/help/contents

Probably dialogs.html and images with a copy of the new button's icon.

File: addOns/requester/src/main/javahelp/help/contents/dialogs.html [hunk 16]
```diff
@@ -94,6 +94,13 @@ <H3>
 	Anti-CSRF extension is enabled.
 	<br>
 
+	<H3>
+		<img src="images/lowercase-header-button.png" width="16" height="16"
+			alt="" />&nbsp; Lowercase Header Names
+	</H3>
+	This will Lowercase all the header names.
```
Comment (thc202) [inline addOns/requester/src/main/javahelp/help/contents/dialogs.html:101]:
```suggestion
	This will lowercase all the header names, making them compliant with HTTP/2.
```

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java (hunk 12 above)
Comment (thc202) [reply addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java:48]: This and following are pending.

Comment (amitpanwar789) [reply addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java]: Sorry for that!, Now its Done.

Comment (thc202) [reply addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java]: No worries, there's also https://github.com/zaproxy/zap-extensions/pull/5149#discussion_r1424328009

File: addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java (hunk 14 above)
Comment (amitpanwar789) [reply addOns/requester/src/test/java/org/zaproxy/addon/requester/internal/HttpPanelSenderUnitTest.java]: done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thank you!

**Comment by amitpanwar789** [comment]
Thank you @kingthorin  @thc202  for all the help, looking forward to make more contribution, if there any project where I can work let me know

**Comment by kingthorin** [comment]
We'd prefer you work on things that interest you but if you're really stuck let us know.
